tests/collections/nlp/test_dialogue.py

```diff
@@ -276,3 +280,224 @@ def test_dialogue_nearest_neighbour_mean_pooling():
+
+    # Test label_id_for_empty_slot to be 0, without PAD
+    # longer empty entity [0, 0, 0]
+    # sentence end with an I-entity [..., 10]
+    slot_label_list = [6, 0, 0, 0, 9, 10, 10, 10, 10]
+    label_id_for_empty_slot = 0
+    expected_output = [1, 0, 0, 0, 1, 1, 2, 2, 2]
+    actual_output = DialogueZeroShotSlotFillingDataset.get_bio_slot_label_from_sequence(
+        slot_label_list, label_id_for_empty_slot
+    )
+    assert expected_output == actual_output
+
+
+@pytest.mark.unit
+def test_dialogue_bert_dataset_get_bio_mention_labels():
+    # Test label_id_for_empty_slot to be 54, without PAD
+    slot_list = [54, 54, 54, 0, 0, 54, 54, 46, 46, 12, 48]
+    bio_list = [0, 0, 0, 1, 2, 0, 0, 1, 2, 1, 1]
+    expected_output = [0, 46, 12, 48, -1, -1, -1, -1, -1, -1, -1]
+    actual_output = DialogueZeroShotSlotFillingDataset.get_bio_mention_labels(slot_list, bio_list)
+    assert expected_output == actual_output
+
+    # Test label_id_for_empty_slot to be 0, with PAD
+    slot_list = [0, 0, 0, 0, 0, 6, 3, 3, 0, 2, 2, -1, -1]
+    bio_list = [0, 0, 0, 0, 0, 1, 1, 2, 0, 1, 2, 0, 0]
+    expected_output = [6, 3, 2, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1]
+    actual_output = DialogueZeroShotSlotFillingDataset.get_bio_mention_labels(slot_list, bio_list)
+    assert expected_output == actual_output
+
+    # Test label_id_for_empty_slot to be 54,without PAD
+    # sentence start with an entity [0, ...]
+    # entity follow by a different entity [0, 1, 2]
+    # an empty entity follow by a B-entity and I-entity [54, 46, 46]
+    # sentence end with an entity [..., 15]
+    slot_list = [0, 1, 2, 54, 46, 46, 54, 12, 15]
+    bio_list = [1, 1, 1, 0, 1, 2, 0, 1, 1]
+    expected_output = [0, 1, 2, 46, 12, 15, -1, -1, -1]
+    actual_output = DialogueZeroShotSlotFillingDataset.get_bio_mention_labels(slot_list, bio_list)
+    assert expected_output == actual_output
+
+    # Test label_id_for_empty_slot to be 54, with PAD
+    # sentence start with an empty entity [54, ...]
+    # an (B-entity, I-entity) follow by a (B-entity, I-entity) [0, 0, 46, 46]
+    # an entity has loger I-entity [46, 46, 46]
+    # sentence end with an empty entity [..., 54]
+    slot_list = [54, 0, 0, 46, 46, 46, 54, 54, -1, -1]
+    bio_list = [0, 1, 2, 1, 2, 2, 0, 0, 0, 0]
+    expected_output = [0, 46, -1, -1, -1, -1, -1, -1, -1, -1]
+    actual_output = DialogueZeroShotSlotFillingDataset.get_bio_mention_labels(slot_list, bio_list)
+    assert expected_output == actual_output
+
+    # Test label_id_for_empty_slot to be 0, without PAD
+    # longer empty entity [0, 0, 0]
+    # sentence end with an I-entity [..., 10]
+    slot_list = [6, 0, 0, 0, 9, 10, 10, 10, 10]
+    bio_list = [1, 0, 0, 0, 1, 1, 2, 2, 2]
+    expected_output = [6, 9, 10, -1, -1, -1, -1, -1, -1]
+    actual_output = DialogueZeroShotSlotFillingDataset.get_bio_mention_labels(slot_list, bio_list)
+    assert expected_output == actual_output
+
+
+@pytest.mark.unit
+def test_align_mention_to_tokens():
```

Comment:
This set of unit tests are so much better than the above! Use this as a guideline to rewrite the above test cases, if you have time.